src/executables/util_cs.c

```diff
@@ -3993,3 +3993,255 @@ tde (UTIL_FUNCTION_ARG * arg)
+   * 1. 10 minutes before end time, if only end time is specified
+   * 2. 10 minutes after start time, if only start time is specified
+   * 3. if no time is specified, then get current execution time for end time */
+
+  if (start_time == 0 && end_time != 0)
+    {
+      start_time = end_time - 600;
+    }
+  else if (start_time != 0 && end_time == 0)
+    {
+      end_time = start_time + 600;
+    }
+  else if (start_time == 0 && end_time == 0)
+    {
+      end_time = time (NULL);
+      start_time = end_time - 600;
+    }
+
+  /* 1. start time < end time
+   * 2. start time is required to be set after db_creation time (server side check)
+   * 3. start time, and end time are required to be set within the log volume range (server side check) */
+  if (start_time > end_time)
+    {
+      fprintf (stderr, "start time(%lld) is larger than end time(%lld)\n", start_time, end_time);
+      goto error_exit;
+    }
+
+  /* output file check */
+  if (output_file == NULL)
+    {
+      outfp = stdout;
+    }
+  else
+    {
+      outfp = fopen (output_file, "w");
+      if (outfp == NULL)
+	{
+	  fprintf (stderr, "can not open output file\n");
+	  goto error_exit;
+	}
+    }
+
+  /* error message log file */
+  snprintf (er_msg_file, sizeof (er_msg_file) - 1, "%s_%s.err", database_name, arg->command_name);
+  er_init (er_msg_file, ER_NEVER_EXIT);
+
+  db_set_client_type (DB_CLIENT_TYPE_ADMIN_UTILITY);
+  if (db_login ("DBA", dba_password) != NO_ERROR)
+    {
+      PRINT_AND_LOG_ERR_MSG ("%s\n", db_error_string (3));
+      goto error_exit;
+    }
+
+  /* first try to restart with the password given (possibly none) */
+  error = db_restart (arg->command_name, TRUE, database_name);
+  if (error)
+    {
+      if (error == ER_AU_INVALID_PASSWORD && (dba_password == NULL || strlen (dba_password) == 0))
+	{
+	  /*
+	   * prompt for a valid password and try again, need a reusable
+	   * password prompter so we can use getpass() on platforms that
+	   * support it.
+	   */
+
+	  /* get password interactively if interactive mode */
+	  /* TODO : MSGCAT setting will be dealt in other issue, it temporarily uses TDE MSGCAT value */
+	  passbuf = getpass (msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_TDE, TDE_MSG_DBA_PASSWORD));
+	  if (passbuf[0] == '\0')	/* to fit into db_login protocol */
+	    {
+	      passbuf = (char *) NULL;
+	    }
+	  dba_password = passbuf;
+	  if (db_login ("DBA", dba_password) != NO_ERROR)
+	    {
+	      PRINT_AND_LOG_ERR_MSG ("%s\n", db_error_string (3));
+	      goto error_exit;
+	    }
+	  else
+	    {
+	      error = db_restart (arg->command_name, TRUE, database_name);
+	    }
+	}
+
+      if (error)
+	{
+	  PRINT_AND_LOG_ERR_MSG ("%s\n", db_error_string (3));
+	  goto error_exit;
+	}
+    }
+
+  db_shutdown ();
+
+  if (darray != NULL)
+    {
+      da_destroy (darray);
+    }
+
+  return EXIT_SUCCESS;
+
+print_flashback_usage:
+  util_log_write_errid (MSGCAT_UTIL_GENERIC_INVALID_ARGUMENT);
```

Comment:
Do you change this to a usages message in the next PR?

Reply:
It would be better to change the label(print_flashback_usage) If you just leave an error.

Reply:
I will change the usage messages in other issue which is related to working on error and help message.

Reply:
OK. You need to think about whether to print usage or output explicit errors.